tests/models/test_amp.py

```diff
@@ -173,3 +174,38 @@ def test_cpu_model_with_amp(tmpdir):
 
     with pytest.raises((MisconfigurationException, ModuleNotFoundError)):
         tpipes.run_model_test(trainer_options, model, on_gpu=False)
+
+
+def test_amp_without_apex(tmpdir):
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    model = EvalModelTemplate()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        amp_backend='apex',
+    )
+    assert trainer.amp_backend is None
+    trainer.fit(model)
+    assert trainer.state == TrainerState.FINISHED
+    assert trainer.dev_debugger.count_events('AMP') == 0
+
+
+@pytest.mark.skipif(not torch.cuda.is_available(), reason="test requires GPU machine")
+def test_amp_with_apex(tmpdir):
+    os.environ['PL_DEV_DEBUG'] = '1'
+
+    model = EvalModelTemplate()
+
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_epochs=1,
+        precision=16,
+        amp_backend='apex',
+        gpus=1,
+    )
+    assert str(trainer.amp_backend) == "AMPType.APEX"
+    trainer.fit(model)
+    assert trainer.state == TrainerState.FINISHED
+    assert trainer.dev_debugger.count_events('AMP') >= 1
```

Comment:
assert the exact number of scaling ops that is expected... this is good error checking in general.